### PR TITLE
Chat - Deprecation warning chatlog override

### PR DESCRIPTION
**Pull request by lusmoura**
- Remove support for `chatlog_override`
- Update tests

### CODE REVIEW

File: cohere/client.py [hunk 1]
```diff
@@ -258,17 +250,33 @@ def chat(
         else:
             return Chat.from_dict(response, query=query, client=self)
 
+    def _format_chat_history(self, chat_history: List[Dict[str, str]]) -> List[Dict[str, str]]:
+        # ensure that the user inputs the updated format while still supporting the old one
+        formatted_chat_history = []
+        for entry in chat_history:
+            text = entry["text"]
+            if "speaker" in entry:
+                speaker = entry["speaker"].capitalize()
```
Comment (sanderland) [inline cohere/client.py:259]: is this optional?
also capitalize() != upper() as suggested by docstring

File: cohere/client.py [hunk 2]
```diff
@@ -258,17 +250,33 @@ def chat(
...
+            else:
+                speaker = entry["user_name"].capitalize()
+            formatted_chat_history.append({"user_name": speaker, "text": text})
```
Comment (sanderland) [inline cohere/client.py:262]: should this be "speaker": speaker ?

File: cohere/client.py [hunk 3]
```diff
@@ -258,17 +250,33 @@ def chat(
...
+            if "text" not in entry or "speaker" not in entry:
+                raise CohereError(message="chat_history must be a list of dicts, each mapping the speaker and text.")
+            if "text" in entry and not isinstance(entry["text"], str):
+                raise CohereError(message="text must be a string in chat_history.")
+            if "speaker" in entry and not isinstance(entry["speaker"], str):
+                raise CohereError(message="speaker must be a string in chat_history.")
+            if "speaker" in entry and entry["speaker"] not in ["USER", "CHATBOT", "SYSTEM"]:
+                raise CohereError(message="speaker must be either 'USER', 'CHATBOT' or 'SYSTEM' in chat_history.")
 
     def _validate_chatlog_override(self, chatlog_override: List[Dict[str, str]]) -> None:
```
Comment (sanderland) [inline cohere/client.py:281]: remove?

File: tests/sync/test_chat.py [hunk 4]
```diff
@@ -168,8 +126,8 @@ def test_chat_history(self):
         prediction = co.chat(
             "Who are you?",
             chat_history=[
-                {"user_name": "User", "text": "Hey!"},
-                {"user_name": "Bot", "text": "Hey! How can I help you?"},
+                {"speaker": "USER", "text": "Hey!"},
+                {"speaker": "CHATBOT", "text": "Hey! How can I help you?"},
```
Comment (sanderland) [inline tests/sync/test_chat.py:130]: please test capitalization logic

File: cohere/client.py [hunk 5]
```diff
@@ -223,18 +211,23 @@ def chat(
...
         """
         if chatlog_override is not None:
             self._validate_chatlog_override(chatlog_override)
+            logger.warning(
+                "The 'chatlog_override' parameter is deprecated and will be removed in a future version of this function. "
+                + "Use 'chat_history' to keep track of the conversation instead.",
+            )
 
         if chat_history is not None:
             self._validate_chat_history(chat_history)
```
Comment (sanderland) [inline cohere/client.py:229]: I think validation checks can be combined with format

**Review by sanderland — CHANGES_REQUESTED** [review]
some suggestions and fixes

**Comment by lusmoura** [comment]
> some suggestions and fixes

@sanderland all the suggestions/fixes were related to the chat_history part of the PR, and after some discussion, we decided to postpone these changes. So I'll remove them an re-request the review.

**Review by sanderland — APPROVED** [review]

**Review by BeatrixCohere — APPROVED** [review]

**Review by BeatrixCohere — APPROVED** [review]

**Pull request merged**